internal/cluster/cluster_test.go

```diff
@@ -2376,22 +2377,26 @@ var _ = Describe("Deregister inactive clusters", func() {
 		db, dbName = common.PrepareTestDB()
 		eventsHandler = events.New(db, logrus.New())
 		dummy := &leader.DummyElector{}
-		state = NewManager(getDefaultConfig(), common.GetTestLog(), db, eventsHandler, nil, mockMetric, nil, dummy, mockOperators, nil, nil)
+		mockS3Client = s3wrapper.NewMockAPI(ctrl)
+		state = NewManager(getDefaultConfig(), common.GetTestLog(), db, eventsHandler, nil, mockMetric, nil, dummy, mockOperators, nil, mockS3Client)
 		c = registerCluster()
 	})
 
 	It("Deregister inactive cluster", func() {
+		mockS3Client.EXPECT().DoesObjectExist(gomock.Any(), gomock.Any()).Return(false, nil).Times(1)
 		Expect(state.DeregisterInactiveCluster(ctx, 10, strfmt.DateTime(time.Now()))).ShouldNot(HaveOccurred())
 		Expect(wasDeregisterd(db, *c.ID)).To(BeTrue())
 	})
 
 	It("Do noting, active cluster", func() {
+		mockS3Client.EXPECT().DoesObjectExist(gomock.Any(), gomock.Any()).Times(0)
 		lastActive := strfmt.DateTime(time.Now().Add(-time.Hour))
 		Expect(state.DeregisterInactiveCluster(ctx, 10, lastActive)).ShouldNot(HaveOccurred())
 		Expect(wasDeregisterd(db, *c.ID)).To(BeFalse())
 	})
 
 	It("Deregister inactive cluster with new clusters", func() {
+		mockS3Client.EXPECT().DoesObjectExist(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
```

Comment:
Looks like we could use `Times(1)` here and in other places. Am I missing something?

I'd rather have an explicit value as it will help us build a more robust test suite.

Reply:
It should be `Times(4)` since `DeregisterInactiveCluster` attempts to deregister clusters in bulks of 100. I'll change it.